### ISSUE TITLE
DSpot fails to run analysis

### ISSUE THREAD

**Issue by borisbaldassari** [STAMP-project/dspot#922]
#### Characteristics
- **Issue Type**: bug
- **Reproducibility**: always
- **Severity**: crash
- **Tool/Service/Component**: dspot-maven 3.0.0
- **Execution Environment**: Debian 9.11

#### Description

Launching dspot on [json-simple](https://github.com/fangyidong/json-simple.git) fails:

```shell
boris@kadath:src$ MAVEN_HOME=~/Applis/apache-maven-3.6.0/ ~/Applis/apache-maven-3.6.0/bin/mvn eu.stamp-project:dspot-maven:amplify-unit-tests -Doutput-path=../
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------< com.googlecode.json-simple:json-simple >---------------
[INFO] Building JSON.simple 1.1.1
[INFO] -------------------------------[ bundle ]-------------------------------
[INFO] 
[INFO] --- dspot-maven:3.0.0:amplify-unit-tests (default-cli) @ json-simple ---
[INFO] Using /media/data/dspot_workspace/projects/fangyidong_json-simple.git/src/pom.xml to run maven.
[INFO] Using /home/boris/Applis/apache-maven-3.6.0 for maven home
[INFO] run maven: /home/boris/Applis/apache-maven-3.6.0/bin/mvn clean test -DskipTests dependency:build-classpath -Dmdep.outputFile=target/dspot/classpath
[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.
[INFO] Cache 'frameworkCache' created in EhcacheManager.
[INFO] Path to runner Classes: /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/org/jacoco/org.jacoco.core/0.7.9/org.jacoco.core-0.7.9.jar
[INFO] /home/boris/.m2/repository/org/ow2/asm/asm-debug-all/5.2/asm-debug-all-5.2.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] /home/boris/.m2/repository/eu/stamp-project/test-runner/2.1.1/test-runner-2.1.1-jar-with-dependencies.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.793 s
[INFO] Finished at: 2019-11-23T16:04:27+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal eu.stamp-project:dspot-maven:3.0.0:amplify-unit-tests (default-cli) on project json-simple: Execution default-cli of goal eu.stamp-project:dspot-maven:3.0.0:amplify-unit-tests failed: Index: 0, Size: 0 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

The complete log, with the -X flag, is attached:

[dspot_json_simple_failed_verbose.txt](https://github.com/STAMP-project/dspot/files/3882596/dspot_json_simple_failed_verbose.txt)

#### Steps to reproduce

Clone json-simple source from https://github.com/fangyidong/json-simple.git

Run:

```shell
MAVEN_HOME=~/Applis/apache-maven-3.6.0/ ~/Applis/apache-maven-3.6.0/bin/mvn eu.stamp-project:dspot-maven:amplify-unit-tests -Doutput-path=.
```

**Comment by danglotb** [comment]
Hello @borisbaldassari 

I identified issues around this use case: usage of `org.junit.framework.Assert` as assertions classes which were not recognized by DSpot, extension of a standard collection such as `java.util.ArrayList` which make the observations of the values ending with errors, when the user does not provide any test method name, the `Dspot-maven` plugin used a wrong default value, etc.

A series of pull requests are coming, I'll ping you I managed to amplify this project.

Best.

**Comment by danglotb** [comment]
Hello @borisbaldassari 

I did manage to successfully amplify `json-simple`:

```txt
Test class that has been amplified: org.json.simple.Test
The original test suite kills 83 mutants
The amplification results with 1 new tests
it kills 9 more mutants

Test class that has been amplified: org.json.simple.parser.YylexTest
The original test suite kills 83 mutants
The amplification results with 0 new tests
it kills 0 more mutants

Test class that has been amplified: org.json.simple.JSONValueTest
The original test suite kills 83 mutants
The amplification results with 0 new tests
it kills 0 more mutants

Test class that has been amplified: org.json.simple.JSONArrayTest
The original test suite kills 83 mutants
The amplification results with 1 new tests
it kills 6 more mutants

[INFO] Writing report in target/dspot/output/report.txt
[INFO] DSpot amplified your test suite without errors. (no errors report will be outputted)
[INFO] The amplification ends up with 4 amplified test methods over 6 test classes.
[INFO] Print org.json.simple.JSONArrayTest.java with 1 amplified test cases in target/dspot/output/
[INFO] Print org.json.simple.JSONArrayTest.java with 1 amplified test cases in target/dspot/output//original/
[INFO] Print org.json.simple.JSONValueTest.java with 0 amplified test cases in target/dspot/output//original/
[INFO] Print org.json.simple.Test.java with 1 amplified test cases in target/dspot/output/
[INFO] Print org.json.simple.Test.java with 1 amplified test cases in target/dspot/output//original/
[INFO] Print org.json.simple.parser.YylexTest.java with 0 amplified test cases in target/dspot/output//original/
[INFO] ------------------------------------------------------------------------
```

Please, retry and tell me if you can also amplify this project.

You will have to use the current SNAPSHOT version, _i.e._ `3.0.1-SNAPSHOT`:

```bash
man eu.stamp-project:dspot-maven:3.0.1-SNAPSHOT:amplify-unit-tests
```

Best.

**Comment by borisbaldassari** [comment]
Hiho @danglotb , thanks for the fix, 3.0.1-SNAPSHOT works for me. 
Do you plan to release it soon? 
Thanks again, cheers!
